lib/internal/bootstrap/loaders.js

```diff
@@ -210,23 +210,21 @@ class NativeModule {
   }
 
   // Used by user-land module loaders to compile and load builtins.
-  compileForPublicLoader(needToSyncExports) {
+  compileForPublicLoader() {
     if (!this.canBeRequiredByUsers) {
       // No code because this is an assertion against bugs
       // eslint-disable-next-line no-restricted-syntax
       throw new Error(`Should not compile ${this.id} for public use`);
     }
     this.compileForInternalLoader();
-    if (needToSyncExports) {
-      if (!this.exportKeys) {
-        // When using --expose-internals, we do not want to reflect the named
-        // exports from core modules as this can trigger unnecessary getters.
-        const internal = this.id.startsWith('internal/');
-        this.exportKeys = internal ? [] : ObjectKeys(this.exports);
-      }
-      this.getESMFacade();
-      this.syncExports();
+    if (!this.exportKeys) {
+      // When using --expose-internals, we do not want to reflect the named
+      // exports from core modules as this can trigger unnecessary getters.
+      const internal = this.id.startsWith('internal/');
```

Comment:
```suggestion
      const internal = StringPrototypeStartsWith(this.id, 'internal/');
```
i realize this is a backport PR, so maybe it should land in master with this change first, but it seems important to avoid brittleness in LTS specifically.